src/emulator/apphosting/config.ts

```diff
@@ -1,44 +1,30 @@
-import { join } from "path";
-import { pathExists } from "fs-extra";
-import { logger } from "./utils";
-import { Emulators } from "../types";
-import { APPHOSTING_BASE_YAML_FILE, APPHOSTING_LOCAL_YAML_FILE } from "../../apphosting/config";
+import { basename } from "path";
+import {
+  APPHOSTING_BASE_YAML_FILE,
+  APPHOSTING_LOCAL_YAML_FILE,
+  discoverConfigsAtBackendRoot,
+  loadConfigForEnvironment,
+} from "../../apphosting/config";
 import { AppHostingYamlConfig } from "../../apphosting/yaml";
 
 /**
  * Loads in apphosting.yaml & apphosting.local.yaml, giving
  * apphosting.local.yaml precedence if present.
  */
-export async function getLocalAppHostingConfiguration(
-  sourceDirectory: string,
-): Promise<AppHostingYamlConfig> {
-  const config: AppHostingYamlConfig = AppHostingYamlConfig.empty();
+export async function getLocalAppHostingConfiguration(cwd: string): Promise<AppHostingYamlConfig> {
+  // Get all apphosting yaml files ignoring the apphosting.local.yaml file
+  const appHostingConfigPaths = discoverConfigsAtBackendRoot(cwd).filter(
+    (path) => !path.endsWith(APPHOSTING_LOCAL_YAML_FILE),
```

Comment:
Seems like we're filtering out the local yaml file here and then trying to find it in in the map below (L28)?